server/services/stitchedTablesCache.test.ts

```diff
@@ -1,190 +1,195 @@
 import S3Client from '../data/s3Client'
-import AnalyticsService, { StitchedTablesCache } from './analyticsService'
+import AnalyticsService from './analyticsService'
 import { AnalyticsError, TableType } from './analyticsServiceTypes'
+import { StitchedTablesCache, MemoryStitchedTablesCache, FileStitchedTablesCache } from './stitchedTablesCache'
 
 jest.mock('@aws-sdk/client-s3')
 jest.mock('../data/s3Client')
 
 describe('StitchedTablesCache', () => {
-  const s3Client = new S3Client({ region: 'eu-west-1', bucket: 'incentives' }) as jest.Mocked<S3Client>
-  const analyticsService = new AnalyticsService(s3Client, () => '')
-  const modified = new Date('2022-03-14T12:00:00Z')
-
-  beforeEach(() => {
-    s3Client.getObject.mockClear()
-    s3Client.listObjects.mockResolvedValue([{ key: 'behaviour_entries/2022-03-13.json', modified }])
-    s3Client.getObject.mockResolvedValue(`{
-      "prison": {"1": "ABC", "2": "DEF"},
-      "someColumn": {"1": "Value 1", "2": "Value 2"}
-    }`)
+  it('makes cache keys', () => {
+    const cache = new MemoryStitchedTablesCache()
+
+    expect(cache.getCacheKey(TableType.incentiveLevels, [])).toEqual(TableType.incentiveLevels)
+    expect(cache.getCacheKey(TableType.behaviourEntries, ['positives', 'negatives'])).toEqual(
+      `${TableType.behaviourEntries},positives,negatives`,
+    )
   })
+})
 
-  describe('clear()', () => {
-    const columnsToStitch = ['prison']
+describe.each([
+  ['in-memory', MemoryStitchedTablesCache],
+  ['file system', FileStitchedTablesCache],
+])('Stitched table caches', (cacheDescription, CacheConstructor) => {
```

Comment:
NB: the exact same test methods are now used for both types of caches; diff is purely whitespace changes